sys/include/net/nanocoap_vfs.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2022 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_nanosock
+ * @brief       VFS NanoCoAP helper functions
+ *
+ * @{
+ *
+ * @file
+ * @brief       VFS NanoCoAP helper functions
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+#ifndef NET_NANOCOAP_VFS_H
+#define NET_NANOCOAP_VFS_H
+
+#include "net/nanocoap_sock.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Downloads the resource behind @p url via blockwise
+ *          GET and stores it in the file @p dst.
+ *
+ * @param[in]   url     URL to the resource
+ * @param[in]   dst     Path to the destination file
+ *
+ * @returns     0 on success
+ * @returns     <0 on error
+ */
+int nanocoap_vfs_get_url(const char *url, const char *dst);
+
+/**
+ * @brief   Downloads the resource behind @p path via blockwise
+ *          GET and stores it in the file @p dst.
+ *
+ * @param[in]   sock    Connection to the server
+ * @param[in]   path    Remote path to the resource
+ * @param[in]   dst     Local path to the destination file
+ *
+ * @returns     0 on success
+ * @returns     <0 on error
+ */
+int nanocoap_vfs_get(nanocoap_sock_t *sock, const char *path, const char *dst);
```

Comment:
Whenever there is a pair like this (get vs. get_url), I wonder when to use which, if the use cases overlap or whether there are combinations that would be needed. (For example, can one use an established socket with a URL that has query components, or a Uri-Host component?)

Here it's probably motivated by the duality of `nanocoap_sock_get_blockwise` and `nanocoap_get_blockwise_url`, where I should have asked the question earlier.

I'd appreciate a note on when to use which, but maybe there are no good answers until https://github.com/RIOT-OS/RIOT/issues/13827 is through.

Reply:
`nanocoap_vfs_get_url()` is just there for convenience to avoid boilerplate in the common case.

The path component (and everything that goes with it) will be parsed the same way for both in `nanocoap_sock_get_blockwise()`.

Reply:
So the path argument is really more path and query? If so, that should be documented (dejavu, I think we've had this somewhere else already.)

Reply:
Can I just call it query path?